unblob/logging.py

```diff
@@ -66,6 +66,31 @@ def add_pid_to_log_message(
     return event_dict
 
 
+def move_exception_to_exc_info(
+    _logger, _method_name: str, event_dict: structlog.types.EventDict
+):
+    """Workaround for a structlog's shortcoming.
+
+    ConsoleRenderer has "special" requirements for keys, e.g. raises an Exception if there
+    is an "exception" key in the passed in event_dict, and its value is an Exception instance.
+
+    Counterintuitively ConsoleReader requires the exception's value to be a string!
+
+    We have some code that builds a dict for logging that includes this "exception" key:
+    e.g. logging report.UnknownError and its derivatives.
+
+    This is necessary a hackish solution, that depends on structlog.processors.format_exc_info
+    to do its work after it.
```

Comment:
Trying to clarify this a little bit:

I can see why are you calling this a "hackish" solution, but in a way it is aligned with the standard library logging. When you log an exception in an except block, under the hood an error is logged with the `exc_info` filed with the exception object: https://github.com/python/cpython/blob/c12ba6b2ff7908c8970b978f149d51ecd3fb195c/Lib/logging/__init__.py#L1521.  Structlog' `ConsoleRenderer` relies on this behavior by getting the `exc_info` and making a pretty string from it and putting it to the `exception` key.

The "problem" with our code is that we call the exception object `exception` rather than `exc_info`, so the pure exception object will be the value of the `exception` key and not the pretty string. We could simple rename the `exception` attribute of the `UnknownError` to `exc_info`, but that maybe would make our code uglier, so I think this (include this into the log processor chain) is the correct way of solving this problem.

Reply:
Another probing question: Why do we need to store exception objects in the first place? Isn't it enough to format the backtrace and store that? An exception has references top all of the associated frame objects and transitively to all variables that were in scope in each of the affected frames. I can see a  possibility of having some hard to detect bugs by keeping exceptions until the end of the program lifetime.